crates/shared/src/price_estimation.rs

```diff
@@ -519,6 +520,27 @@ pub fn amounts_to_price(sell_amount: U256, buy_amount: U256) -> Option<BigRation
     ))
 }
 
+#[derive(prometheus_metric_storage::MetricStorage)]
+pub(in crate::price_estimation) struct Metrics {
+    /// number of requests done by each estimator
+    #[metric(labels("requests"))]
+    native_price_requests: IntCounterVec,
+}
+
+impl Metrics {
+    fn get() -> &'static Metrics {
+        Metrics::instance(observe::metrics::get_storage_registry())
+            .expect("unexpected error getting metrics instance")
+    }
+
+    pub(in crate::price_estimation) fn inc_estimator(estimator: &str) {
```

Comment:
I think the method on a `Metrics` struct should ideally describe what it tries to track instead of what it does internally. So in this case `Metrics::sent_request("oneinch")` for example.